### PR TITLE
Expose WebHostBuilderContext in UseDefaultServiceProvider

### PR DESCRIPTION
**Pull request by pakrym**
As part of https://github.com/aspnet/DependencyInjection/issues/481

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs [hunk 1]
```diff
@@ -86,12 +86,12 @@ public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Type
         /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
         /// <param name="configure">A callback used to configure the <see cref="ServiceProviderOptions"/> for the default <see cref="IServiceProvider"/>.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
-        public static IWebHostBuilder UseDefaultServiceProvider(this IWebHostBuilder hostBuilder, Action<ServiceProviderOptions> configure)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs]: Add both overloads to avoid breaking people? Or is this a new 2.0 API?

Comment (pakrym) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs]: Was added in https://github.com/aspnet/Hosting/commit/d57d729d13c73750027944dda25082693d6b6d7f it's new in 2.0

**Comment by davidfowl** [comment]
@pakrym Add both overloads

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs]: Add the overload either way.

**Pull request merged**